### DISCUSSION TITLE
Can Saleor use products created on the fly? So pulled in from our headless CMS and not added into Saleor?

### DISCUSSION THREAD

**Discussion by amityweb** [saleor/saleor#9555] [General]
Our customer wants to use one headless CMS for all data, Strapi. All products need to be in there. We don't really want to be also adding products in Saleor, and then linking them with IDs, thats not good. 

So can we submit orders and product info to Saleor on the fly? So therefore we can build our products from the Strapi data, and build in an add to cart etc. and on checkout it submits the product data on the fly to Saleor. The products wont be in Saleor. 

I read Snipcart could do this, or Medusajs can auto sync the product data so no need to manually create it. Any suggestions for using Saleor without manually having to create the duplicate product data would be welcome. 

Thanks

**Comment by karolkielecki** [comment]
Hi @amityweb, I'm afraid that's not possible. Products should be added on the Saleor's side.
